### PR TITLE
Wip plotting

### PR DESCRIPTION
**Pull request by ysimillides**
started the plotting functionality, but still allows FEniCS to be run without it

### CODE REVIEW

File: src/FEniCS.jl [hunk 1]
```diff
@@ -34,11 +35,19 @@ include("jmesh.jl") #this file contains the mesh functions
...
-#include("jplot.jl") #this file contains the plotting functionality
+include("jinterface.jl")
 try
   pyimport("mshr")
   include("fmshr.jl") #this file contains various geometrical objects using the mshr package
 catch ee
  print("mshr has not been included")
 end
+try
+  Pkg.installed("PyPlot")
```
Comment (ChrisRackauckas) [inline src/FEniCS.jl]: instead, use Requires.jl

File: REQUIRE [hunk 2]
```diff
@@ -1,3 +1,4 @@
 julia 0.6
 PyCall
 Conda
+PyPlot
```
Comment (ChrisRackauckas) [inline REQUIRE]: This isn't necessary with Requires.jl

File: src/FEniCS.jl [hunk 3]
```diff
@@ -1,9 +1,10 @@
+__precompile__(false)
```
Comment (ChrisRackauckas) [inline src/FEniCS.jl]: precompilation should work with Requires

File: src/jinterface.jl [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+Creates a type which incorporates the mesh attributes alongside the necessary values
+to solve finite element problems. It works by taking in a FEniCS mesh, working out
+the necessary values, and then creating an ordering of the internal/boundary nodes
+and applying the DirichletBC (via a function) to the respective nodes.
+"""
+type feMesh
+    n_nodes::Int
+    n_elements::Int
+    nodes::Array{Float64, 2}
+    internal_nodes::Vector
```
Comment (ChrisRackauckas) [inline src/jinterface.jl]: `Vector{Float64}`

**Comment by ysimillides** [comment]
subset of #49 , so closing this one

**Comment by ChrisRackauckas** [comment]
The plotting should be a different PR. You can do this by making the PRs from different branches so they don't overlap.

**Comment by ChrisRackauckas** [comment]
Were all of the changes mentioned in the comments here added to the new PR?

**Pull request closed without merging**